### PR TITLE
Fix restmapper

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/7090

The `MultiRESTMapper` didn't correctly combine errors and the shortcut expanding RESTMapper failed to override the required functions because it was anonymously including another RESTMapper.

New message for missing resource:
```
[deads@deads-dev-01 origin]$ oc get foo
the server doesn't have a resource type "foo"
```

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/meta/multirestmapper.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+func (m MultiRESTMapper) String() string {
+	nested := []string{}
+	for _, t := range m {
+		currString := fmt.Sprintf("%v", t)
+		splitStrings := strings.Split(currString, "\n")
+		nested = append(nested, strings.Join(splitStrings, "\n\t"))
+	}
+
+	return fmt.Sprintf("MultiRESTMapper{\n\t%s\n}", strings.Join(nested, "\n\t"))
```
Comment (kargakis) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/meta/multirestmapper.go]: Is this for debugging?

Comment (kargakis) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/meta/multirestmapper.go]: Oh, I see it's just refactoring of old code. Nvmd

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource/builder.go [hunk 2]
```diff
@@ -440,15 +440,27 @@ func (b *Builder) resourceMappings() ([]*meta.RESTMapping, error) {
...
+		gvks, err := b.mapper.KindsFor(unversioned.GroupVersionResource{Resource: r})
 		if err != nil {
 			return nil, err
 		}
-		mapping, err := b.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
-		if err != nil {
-			return nil, err
+		// the list is in most-preferred to least preferred, so iterate until we find a hit
+		var firstErr error
+		for _, gvk := range gvks {
```
Comment (kargakis) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource/builder.go]: Could this be another method in the restmapper?

Comment (deads2k) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource/builder.go]: > Could this be another method in the restmapper?

Maybe, but not in this pull.  It seems like someone wants a `RESTMappingForPartialResource`, but should still give back a list.

**Comment by kargakis** [comment]
LGTM

**Comment by deads2k** [comment]
@kargakis `test-cmd` caught a pre-existing bug in the resource match when I tidied up the resource detection for arguments.  Two more commits to look at.

**Comment by deads2k** [comment]
[merge]

**Comment by deads2k** [comment]
[merge]

**Pull request merged**